backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1758,3 +1758,85 @@ databaseChangeLog:
             columns:
               - column:
                   name: test_result_delivery
+  - changeSet:
+      id: add-no-phi-views
+      author: josh@skylight.digital
+      comment: Add views for some tables that omit columns containing PHI.
+      changes:
+        - createView:
+            viewName: api_user_no_phi_view
+            remarks: A subset of the api_user table with columns containing PHI removed.
+            replaceIfExists: true
+            fullDefinition: false
+            selectQuery: SELECT internal_id, created_at, updated_at, last_seen, is_deleted FROM ${database.defaultSchemaName}.api_user
+        - createView:
+            viewName: provider_no_phi_view
+            remarks: A subset of the provider table with columns containing PHI removed.
+            replaceIfExists: true
+            fullDefinition: false
+            selectQuery: SELECT internal_id, created_at, created_by, updated_at, updated_by, is_deleted, provider_id FROM ${database.defaultSchemaName}.provider
+        - createView:
+            viewName: person_no_phi_view
+            remarks: A subset of the person table with columns containing PHI removed.
+            replaceIfExists: true
+            fullDefinition: false
+            selectQuery: SELECT internal_id, created_at, created_by, updated_at, updated_by, is_deleted, organization_id, race, gender, ethnicity, lookup_id, county, state, employed_in_healthcare, role, resident_congregate_setting, facility_id, tribal_affiliation FROM ${database.defaultSchemaName}.person
+        - createView:
+            viewName: test_event_no_phi_view
+            remarks: A subset of the test_event table with columns containing PHI removed.
+            replaceIfExists: true
+            fullDefinition: false
+            selectQuery: SELECT device_specimen_type_id, created_at, created_by, updated_at, updated_by, patient_id, organization_id, device_type_id, result, facility_id, survey_data, date_tested_backdate, test_order_id, correction_status, prior_corrected_test_event_id, internal_id, reason_for_correction FROM ${database.defaultSchemaName}.test_event
+      rollback:
+        - dropView:
+            viewName: api_user_no_phi_view
+        - dropView:
+            viewName: provider_no_phi_view
+        - dropView:
+            viewName: person_no_phi_view
+        - dropView:
+            viewName: test_event_no_phi_view
+  - changeSet:
+      id: set-no-phi-user-permissions
+      author: josh@skylight.digital
+      comment: Restrict access for the simple_report_no_phi user to certain tables/columns in order to prevent PHI exposure.
+      changes:
+        - sql:
+          sql: |
+            GRANT USAGE ON SCHEMA ${database.defaultSchemaName} TO simple_report_no_phi;
+            GRANT SELECT ON TABLE ${database.defaultSchemaName}.facility_device_type TO simple_report_no_phi;
```

Comment:
This I think this changeset needs:

* a changelog property for the user, rather than having it hard-coded
* an execution condition such that if the user property is not set, it doesn't try to run the migration

The combination should let us merge this and have it run in test and prod but not in other environments until/unless we get the user-creation setup generalized to all environments.

Reply:
I think this should run in all environments, actually, for the sake of uniformity. It's fine if it's unused, but I think we should avoid intentionally introducing any more environment differences.

Reply:
I support it running in all environments, but I also support having a safety-valve mechanism for any external dependency such as database user creation—I think the precondition gives us that, pretty cheaply.

Reply:
Can you say more about a safety-valve mechanism? What would we be protecting against?

Also, parameterizing the user is probably a bad idea since we need this user to match what's in Terraform.

Reply:
Outcome of offline discussion: let's parameterize the username, if only to DRY it up, and run this in all environments.